addons/pos_online_payment/controllers/payment_portal.py

```diff
@@ -29,10 +29,7 @@ def _ensure_session_open(pos_order_sudo):
             raise AccessError(_("The POS session is not opened."))
 
     def _get_partner_sudo(self, user_sudo):
-        partner_sudo = user_sudo.partner_id
-        if not partner_sudo and user_sudo._is_public():
-            partner_sudo = self.env.ref('base.public_user')
```

Comment:
Can't you accomplish this fix by changing this line from

`partner_sudo = self.env.ref('base.public_user')`

to 

`partner_sudo = self.env.ref('base.public_partner')`?

Reply:
@caburj 

- Can you please check the condition before that, `if not partner_sudo` which has no sense like every user has their associated partner if I'm not wrong > So there is no case where condition is satisfied
- Second thing is self won't work with controller so If by mistake the condition will satisfy then the error will come.

Reply:
Additionally, this error will occur when the order does not have an associated partner. Simply changing `self.env.ref('base.public_user')` to `request.env.ref('base.public_partner')` won't resolve the issue because, in that case, the user will be the logged-in user, and the partner will be the logged-in user's partner.